client/my-sites/plans-grid/lib/get-plan-pricing-info-from-grid-plans.ts

```diff
@@ -0,0 +1,31 @@
+import type { GridPlan } from 'calypso/my-sites/plans-grid/hooks/npm-ready/data-store/use-grid-plans';
+
+type PricingInfo = {
+	prices: number[];
+	currencyCode?: string;
+};
+
+export function getPlanPricingInfoFromGridPlans( {
+	gridPlans,
+	returnMonthly = true,
+}: {
+	gridPlans: GridPlan[];
+	returnMonthly?: boolean;
+} ) {
+	return gridPlans.reduce(
```

Comment:
I think we should be memoizing the return value to (1) avoid recomputing, and (2) avoid re-rendering anything that consumes it (React shallow compares props to decide). 

To go about this, I'd recommend turning`getPlanPricingInfoFromGridPlans` into a hook and wrapping the return value with `useMemo` (from `@wordpress/element`).

Reply:
> I'd recommend turning `getPlanPricingInfoFromGridPlans` into a hook

This should have been the criteria from the start IMO. to opt for hook-like **exported** functions and make room for their use, rather than selector-like functions (unless necessary i.e. no other way). We want uniformity in these data helpers -> hence be able to optimize uniformly, force uniform component structures that consume them (e.g. function components), and be able to unify inside a store later.

p.s. the only other selector-like getter that exists in `pland-grid/lib` is something that was ported from elsewhere, and is subject to removal :-)

Reply:
> to opt for hook-like exported functions and make room for their use, rather than selector-like functions (unless necessary i.e. no other way).

I wasn't aware of this, but thanks for letting me know. I'll be sure to keep this in mind for future development in the plans pricing package 👍 

Resolved in https://github.com/Automattic/wp-calypso/pull/81743/commits/03be67f319d6199eedb739527072aaaadecdfd54